src/ChannelServer/Network/ChannelClient.cs

```diff
@@ -104,9 +104,13 @@ public Creature GetSummonedPetSafe(long id)
 		/// </summary>
 		public override void CleanUp()
 		{
-			// Dispose creatures, to remove subscriptions and stuff.
-			// Do this before unspawning, the creature might need the region.
-			foreach (var creature in this.Creatures.Values)
+
+            // Moved here to always be called when a client is being killed off, on the suggestion of E.
+            ChannelServer.Instance.Events.OnPlayerDisconnect(Controlling);
```

Comment:
Hm... maybe this should be called for every creature? What if you want to do something with this event that should happen for all controlled creatures?